File: apple/Sources/ReactNativeWatchOS/Modifiers/Bold.swift
import SwiftUI

/// SwiftUI `.bold(_:)`. Applies a bold font weight to text within the
/// view. Defaults to active when `value` is omitted.
enum RNWBoldModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("bold") { view, params, _ in
            AnyView(view.bold(params.bool("value") ?? true))
        }
    }
}

File: src/modifiers/bold.ts
import { createModifier } from './createModifier';

export interface BoldParams {
  /// Whether bold is active. Defaults to `true`.
  value?: boolean;
}

/// SwiftUI `.bold(_:)`. Applies a bold font weight to text within the
/// view. Pass `false` to explicitly turn it off.
export function bold(params?: BoldParams | boolean) {
  if (typeof params === 'boolean') {
    return createModifier('bold', { value: params });
  }
  return createModifier('bold', params);
}
